enterprise/server/auth/auth.go

```diff
@@ -252,16 +253,21 @@ func (a *authenticator) verifyTokenAndExtractUser(ctx context.Context, jwt strin
 }
 
 type OpenIDAuthenticator struct {
-	env            environment.Env
-	myURL          *url.URL
-	authenticators []*authenticator
+	env              environment.Env
+	myURL            *url.URL
+	authenticators   []*authenticator
+	apiKeyGroupCache *ttlcache.Cache
```

Comment:
could we replace this with our own util/lru cache?

It has the advantage of:
 - being a fixed configurable size (lets us reason about how big our machines need to be)
 - is trusted code that we're confident storing things like this in

We can store structs that have an expiry time in them, and if on fetch, if it's out of date, you can just drop it. You could either do that in your code here or write a TTL wrapper that uses the lru cache under the hood.

Reply:
Done.

Note that I also changed the config to enable the caching by default with an option to disable it.